AWSTranscribeStreaming/Internal/AWSSRWebSocketDelegateAdaptor.m

```diff
@@ -88,7 +92,7 @@ - (void)webSocketDidOpen:(AWSSRWebSocket *)webSocket {
  @param webSocket the web socket that failed
  @param error the error causing the failure
  */
-- (void)webSocket:(AWSSRWebSocket *)webSocket didFailWithError:(NSError *)error {
+- (void)webSocket:(nullable AWSSRWebSocket *)webSocket didError:(NSError *)error {
```

Comment:
As above, why do we need the extra layer of indirection?

Reply:
cuz of the socket rocket protocol. if method names match, how will the adaptor class know which one to call when one has to call the other?

Reply:
I may be misunderstanding your point. We register this adaptor as the SocketRocket delegate. SR will invoke `-[webSocket:didFailWithError:]` on its delegate, which will then foward that error to the AWSTranscribeStreamingDelegate. What do you mean by "when one has to call the other"?

Reply:
exactly what you said. SR invokes `didFailWithError` on its delegate which inside the same adaptor class calls the AWSTranscribeStreamingDelegate function `didError`

Reply:
```
- (void)webSocket:(AWSSRWebSocket *)webSocket didFailWithError:(NSError *)error {
    [self webSocket:webSocket didError:error];
}
```

Reply:
if this was 
```
- (void)webSocket:(AWSSRWebSocket *)webSocket didFailWithError:(NSError *)error {
    [self webSocket:webSocket didFailWithError:error];
}
```
how would the class know which function I am calling when they look exactly the same?